### DISCUSSION TITLE
Can I skip axios interceptors.request ?

### DISCUSSION THREAD

**Discussion by okinawaa** [mswjs/msw#1638] [Q&A]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

Inside my component, I am using axios' interceptors.request.use

I hope this interceptors syntax doesn't work for axios calls via `msw` inside the storybook.

Is there a way to skip interceptors?

### Example

```tsx
const authApi: AxiosInstance = axios.create(...);

authApi.interceptors.request.use(
  async (config): Promise => await requestProcess(config) // I don't want excute this function in storybook 
);

```

**Comment by kettanaito** [comment]
Hi, @ChanhyukPark-Tech. 

I don't think this is something MSW should do. The [Axios interceptors](https://axios-http.com/docs/interceptors) is a feature of Axios so you should consult Axios documentation on how to skip particular interceptors in scenarios like your Storybook stories. 

I think MSW should work without issues with Axios interceptors API. You can intercept any requests your Axios client makes and decide how to handle them. MSW on its own doesn't meddle with your traffic so it has no control over _excluding_ certain requests on the page—that's out of the library's scope. What you can do is capture that unwanted request in MSW and respond with a mock response to it. That translates to a request that happened but was mocked. 

Also, please note that Axios interceptors documentation already lists the way to remove a certain interceptor from execution:

```js
const myInterceptor = axios.interceptors.request.use(function () {/*...*/});
axios.interceptors.request.eject(myInterceptor);
```

**Reply by okinawaa** [reply]
Thank you very much for your kind reply. 
Once again, I have learned the role range of the library.
Thank you for suggesting a good solution 
I'll continue to cheer for you, MSW!
